### PR TITLE
Initial

### PR DESCRIPTION
**Pull request by scribe**

### CODE REVIEW

File: guava/src/main/kotlin/com/spectralogic/kotlinutils/core/curry.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.spectralogic.kotlinutils.core
```
Comment (rpmoore) [inline guava/src/main/kotlin/com/spectralogic/kotlinutils/core/curry.kt]: The filename for this should be Capitalized.

File: guava/src/main/kotlin/com/spectralogic/kotlinutils/core/curry.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+/*
+ Composible curry function for reducing the arity of a given function
+ */
+
+fun <A,B> ((A) -> B).curry(a: A): B = this.invoke(a)
```
Comment (rpmoore) [inline guava/src/main/kotlin/com/spectralogic/kotlinutils/core/curry.kt]: Include some Javadoc for each of these curry functions to better explain what the different versions do.

File: guava/src/main/kotlin/com/spectralogic/kotlinutils/core/curry.kt (hunk 1 above)
Comment (rpmoore) [reply guava/src/main/kotlin/com/spectralogic/kotlinutils/core/curry.kt]: Also, should this file be in the guava module, or in a core module?

File: guava/src/test/kotlin/com/spectralogic/kotlinutils/core/curryTest.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package com.spectralogic.kotlinutils.core
```
Comment (rpmoore) [inline guava/src/test/kotlin/com/spectralogic/kotlinutils/core/curryTest.kt]: Capitalize this filename

File: guava/src/main/kotlin/com/spectralogic/kotlinutils/guava/Memoize.kt [hunk 4]
```diff
@@ -0,0 +1,41 @@
+package com.spectralogic.kotlinutils.guava
+
+import com.google.common.cache.Cache
+import com.google.common.cache.CacheBuilder
+
+fun <A> (() -> A).memoize(memoSize: Long = 1000L): () -> A {
```
Comment (rpmoore) [inline guava/src/main/kotlin/com/spectralogic/kotlinutils/guava/Memoize.kt]: Add Javadoc to explain how these should be used.  Provide some usage examples as well.

**Review by rpmoore — CHANGES_REQUESTED** [review]

File: guava/src/main/kotlin/com/spectralogic/kotlinutils/core/curry.kt (hunk 1 above)
Comment (scribe) [reply guava/src/main/kotlin/com/spectralogic/kotlinutils/core/curry.kt]: My thinking was that curry doesn't rely on Guava, so it should be in core, but I can move it.

**Review by rpmoore — APPROVED** [review]

**Pull request merged**